main.py

```diff
@@ -120,20 +149,64 @@ def update_board(board):
     return board
 
 
+def draw_fps_slider(y, fps_limit):
+    """Draws the slider with the y coordinate of the button click
+       (How many FPS this corresponds to is not dealt with here.)"""
+    c = config.Size * config.Width + config.Edge / 2 + config.ButtonSize / 2  # The middle of the line of buttons
+    d = config.StartOfSlider
+    e = config.EndOfSlider
+    if y < d:
+        y = d
+    elif y > e:
+        y = e
+    pygame.draw.rect(Screen, config.Background, ((c + config.NotchLength / 2, d - config.NotchLength),
+                                                 (config.Width * config.Size + config.Edge + config.Border +
+                                                  config.ButtonSize, e)))
+    pygame.draw.line(Screen, (180, 180, 180), (c, d), (c, e))
+    f = (e - d) / config.Notches  # How far each notch goes
+    for g in range(config.Notches + 1):
+        pygame.draw.line(Screen, (180, 180, 180), (c - config.NotchLength / 2, d + g * f),
+                         (c + config.NotchLength / 2, d + g * f))
+    config.write(Screen, c - (12 + config.NotchLength), (d + e) * 0.5, "Speed", (180, 180, 180), 20,
+                 rotate=90, alignment=("left", "centre"))
+    if fps_limit:
+        colour = (0, 255, 100)
+    else:
+        colour = (160, 160, 160)
+    pygame.draw.polygon(Screen, colour, ((c + config.NotchLength / 2, y),
+                                         (c + config.NotchLength, y - config.NotchLength / 2),
+                                         (c + 2 * config.NotchLength, y - config.NotchLength / 2),
+                                         (c + 2 * config.NotchLength, y + config.NotchLength / 2),
+                                         (c + config.NotchLength, y + config.NotchLength / 2)))
+
+
 pygame.init()
-Screen = pygame.display.set_mode((config.Size * config.Width, config.Size * config.Height))
+Screen = pygame.display.set_mode((config.Size * config.Width + config.ButtonSize, config.Size * config.Height))
 Screen.fill(config.Background)
-pygame.display.set_caption("Game of Life")
 Board = clean_board()
 Paused = True
+FPS = 10
```

Comment:
Why has `FPS` been redefined here again? Wasn't this defined in the config?

Reply:
I got it. No need for a variable here https://www.pygame.org/docs/ref/time.html#pygame.time.Clock.get_fps

Reply:
I tried that but it pauses the game at that point, like the time.sleep() function. This means that an input cannot be obtained between frames

Reply:
rename to generations_per_second

Reply:
done